File: core/src/main/java/com/example/coinstest/data/CurrencyInRubDataSource.java
package com.example.coinstest.data;

import com.example.coinstest.domain.CurrencyInRub;

import java.time.LocalDate;

public interface CurrencyInRubDataSource {
    CurrencyInRub getCurrencyOnDate(LocalDate date);

    LocalDate getLastDate();
}

File: core/src/main/java/com/andysklyarov/finnotifyfree/data/CurrencyInRubDataSource.kt
package com.andysklyarov.finnotifyfree.data

import com.andysklyarov.finnotifyfree.domain.CurrencyInRub
import io.reactivex.Single
import java.time.LocalDate

interface CurrencyInRubDataSource {
    fun getCurrencyOnDate(currencyCode: String, date: LocalDate): Single<CurrencyInRub>
    fun getCurrencyOnPreviousDate(currencyCode: String, date: LocalDate): Single<CurrencyInRub>
    fun getLastDate(): Single<LocalDate>
}